text/0000-or-patterns.md

```diff
@@ -387,27 +387,83 @@ want the same outcome as `foo @ 1 | foo @ 2 | foo @ 3`, you have to write
 `foo @ (1 | 2 | 3)` instead of writing `foo @ 1 | 2 | 3`.
 This is discussed in the [rationale][alternatives].
 
+You can also use `p | q` in:
+
+1. `if let` expressions:
+
+   ```rust
+   if let Foo::Bar | Foo::Quux(1 | 2) = some_computation() {
+       ...
+   }
+   ```
+
+1. `while let` expressions:
+
+   ```rust
+   while let Ok(1 | 2) | Err(3) = different_computation() {
+       ...
+   }
+   ```
+
+3. `let` statements:
+
+   ```rust
+   let Ok(x) | Err(x) = another_computation();
+   ```
+
+   In this case, the pattern must be irrefutable as `Ok(x) | Err(x)` is.
+
+4. `fn` arguments:
+
+   ```rust
+   fn foo(Ok(x) | Err(x): Result<u8, u8>) {
```

Comment:
I personally think this should use `no_top_alt` as well, to improve readability.